app/controllers/admin/base_controller.rb

```diff
@@ -1,5 +1,16 @@
 # frozen_string_literal: true
 
 class Admin::BaseController < ApplicationController
+  include Pundit
+
   before_action :authenticate_user!
+  before_action :ensure_admin
+  after_action :verify_authorized
+  after_action :verify_policy_scoped
```

Comment:
Hmm, why do we ensure a user is admin before action, but verify that they are authorised and scoped after?

Reply:
I wasn't sure about this, but these do fundamentally different things. For example, if I'm not an admin, without the `ensure_admin` I'll still be able to index lead providers by going to /admin/lead_providers. Pundit controls models, not pages, so I just wouldn't be able to load the /new or /edit pages.

The alternative is to use headless policies to achieve this, but that seemed a little excessive

Reply:
For example, for the admin dashboard, we could have something like
```
# app/policies/dashboard_policy.rb
class Admin::DashboardPolicy < Struct.new(:user, :dashboard)
  def show?
    return true if user.admin?

    false
  end
end
```
and
```
# app/controllers/admin/dashboard_controller.rb
...
def show
   authorize([:admin, :dashboard], :show?)
end
```
which would do exactly the same thing, but through pundit

Reply:
Alright, makes sense.